src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderBooking.kt

```diff
@@ -0,0 +1,208 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.BatchSize
+import org.hibernate.annotations.Type
+import java.time.LocalDateTime
+import java.util.Objects
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.FetchType
+import javax.persistence.GeneratedValue
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.ManyToOne
+import javax.persistence.SequenceGenerator
+import javax.persistence.Table
+
+@Entity
+@Table(name = "OFFENDER_BOOKINGS")
+@BatchSize(size = 25)
+data class OffenderBooking(
+  @SequenceGenerator(name = "OFFENDER_BOOK_ID", sequenceName = "OFFENDER_BOOK_ID", allocationSize = 1)
+  @GeneratedValue(generator = "OFFENDER_BOOK_ID")
+  @Id
+  @Column(name = "OFFENDER_BOOK_ID")
+  val bookingId: Long? = null,
+
+  @Column(name = "BOOKING_SEQ", nullable = false)
+  val bookingSequence: Int? = null,
+
+  @Column(name = "BOOKING_NO")
+  val bookNumber: String? = null,
+
+  @Column(name = "BOOKING_TYPE")
+  val bookingType: String? = null,
+
+  @ManyToOne(optional = false, fetch = FetchType.LAZY)
+  @JoinColumn(name = "OFFENDER_ID", nullable = false)
+  val offender: Offender? = null,
+
+  @Column(name = "DISCLOSURE_FLAG", nullable = false)
+  val disclosureFlag: String = "Y",
+
+  @Column(name = "BOOKING_BEGIN_DATE", nullable = false)
+  val bookingBeginDate: LocalDateTime,
+
+  //
+  //    @OneToMany(mappedBy = "id.offenderBooking", cascade = CascadeType.ALL)
+  //    @Default
+  //    private List<OffenderProfileDetail> profileDetails = new ArrayList<>();
```

Comment:
Wonder if we'll ever be able factor this common data into a library of some kind?